db/post_migrate/20240918170954_remove_settings_thing.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+class RemoveSettingsThing < ActiveRecord::Migration[7.1]
+  disable_ddl_transaction!
+
+  def change
+    connection.execute(<<~SQL.squish)
+      DELETE FROM settings WHERE thing_type IS NOT NULL and thing_id IS NOT NULL
+    SQL
```

Comment:
I think this needs to be moved to a pre-deployment migration. And now that we have migration breakpoints, it should be safe to do so.

Reply:
Will do -- just to confirm, move **JUST** the delete line to new migration in `db/migrate`, but leave the rest (add index, remove index, remove columns) as-is in this `db/post_migrate` one?

Reply:
That is a good question. I guess the assumption now is that anyone running those will be on 4.3.0 or newer, or will have the services stopped while running the migrations. This means they should normally not lead to new entries with `thing_id` and `thing_type`. Just to be safe we could do the deletion in both a pre and post-deployment migration.

The columns removal still need to be in a post-deployment migration because the 4.3.0 code does not ignore those columns. Index changes should be safe either way I suppose, but if we are re-deleting stuff out of cautiousness in the post-deployment migrations, this means we would do the index changes there too.

Reply:
Rebased, pushed another update...

- Remove previous single migration doing everything
- Added new db/migrate migration to do just the delete
- Added new db/post_migrate to repeat the delete and do the other ops
- Other changes are same

Converted to ready for review as well.